### ISSUE TITLE
`setParams` of a parentNavigator retreived with dangerouslyGetParent doesn't work anymore since 2.3.0

### ISSUE THREAD

**Issue by xanderberkein** [react-navigation/react-navigation#4497]
### Current Behavior

Trying to update a param of the parentNavigator (which I retrieved with dangerouslyGetParent) with `setParams` doesn't work in the latest version of `react-navigation`. See this snack for a demo: https://snack.expo.io/@x11n/react-navigation-setparams

Clicking the buttons should update the title of the ParentNavigator, but nothing happens. If you go to `package.json` and try changing react-navigation to `"react-navigation": "2.2.5"`, everything works fine.

### Expected Behavior

I expect `setParams` to update the params of a parentNavigator

### How to reproduce

See snack: https://snack.expo.io/@x11n/react-navigation-setparams.

### Your Environment

| software         | version
| ---------------- | -------
| react-navigation | 2.3.0 and up

**Comment by brentvatne** [comment]
thanks for sharing, cc @ericvicenti

**Comment by xanderberkein** [comment]
Any updates on this @brentvatne?  
This commit https://github.com/react-navigation/react-navigation/commit/470eaf3b0809227d3e0cf2e67a659aae5a6014cc seems to be the cause of this issue, but I don't have the knowledge of the internals of react-navigation to understand what goes wrong.

**Comment by brentvatne** [comment]
oh yeah, I see why this is happening. in 470eaf3b0809227d3e0cf2e67a659aae5a6014cc we made it so the navigation prop on navigators is the same as the children to fix a variety of usability issues. this generally does the right thing but in this case is a little confusing because you expect when you call `setParams` and then `getParam` on a navigator object that you will get the result from `setParams`. so i will fix that.

one question for you here - if you were to call `parentNavigator.goBack()` what would you expect to happen? similarly, what would you expect to happen if we added another screen to `ChildNavigator` and then added this to the example?

```
ChildNavigator.navigationOptions = ({ navigation }) => ({
  headerRight: <Button title="Go back!" onPress={() => navigation.goBack()} />,
});
```

**Comment by brentvatne** [comment]
see react-navigation@2.5.3

**Comment by xanderberkein** [comment]
Thanks for the fix!

To answer your question, if `parentNavigator.goBack()` would be called, maybe it would make sense to make the `parentNavigator` goBack (and skip the eventual routes of the childNavigator), since you are specifically targetting the parent.   
This seems like an edge case though, in most cases you wouldn't want that behavior and you'd just call `goBack()` on the child.

**Comment by reactdevrocks** [comment]
not working for me but how you fix it @xanderberkein . please help me.

**Comment by xanderberkein** [comment]
@reactdevrocks This issue was fixed in react-navigation@2.5.3. 

If you think this bug popped up again, you can create a new issue with a reproducible example / [snack](https://snack.expo.io/). If you have a question about how to implement `setParams`, Stack Overflow is better suited for this.
